receiver/prometheusreceiver/internal/metrics_adjuster.go

```diff
@@ -42,46 +104,87 @@ func getSignature(name string, values []*metricspb.LabelValue) string {
 	return fmt.Sprintf("%s,%s", name, strings.Join(labelValues, ","))
 }
 
-// JobsMap maps from a job instance to a map of metric instances for the job.
-type JobsMap map[string]*metricsInstanceMap
+// JobsMap maps from a job instance to a map of timeseries instances for the job.
+type JobsMap struct {
+	sync.RWMutex
+	gcInterval time.Duration
+	lastGC     time.Time
+	jobsMap    map[string]*timeseriesMap
+}
 
 // NewJobsMap creates a new (empty) JobsMap.
-func NewJobsMap() *JobsMap {
-	jm := JobsMap(make(map[string]*metricsInstanceMap))
-	return &jm
+func NewJobsMap(gcInterval time.Duration) *JobsMap {
+	return &JobsMap{gcInterval: gcInterval, lastGC: time.Now(), jobsMap: make(map[string]*timeseriesMap)}
 }
 
-func (jm *JobsMap) get(job, instance string) *metricsInstanceMap {
+// Remove jobs and timeseries that have aged out.
+func (jm *JobsMap) gc() {
+	current := time.Now()
+	jm.Lock()
+	defer jm.Unlock()
+	// recheck lastGC to make sure gc is necessary
+	if current.Sub(jm.lastGC) > jm.gcInterval {
```

Comment:
This seems duplicated to https://github.com/census-instrumentation/opencensus-service/pull/613/files#diff-6d65923b82ce96fdb3462374e44dd2e8R140.

Reply:
Yes, the previous check is done before locking jm - this check is to verify that no other gc() has happened in the interim before we grab the lock. I'll add a comment to clarify what's going on.